libpromises/evalfunction.c

```diff
@@ -5473,9 +5475,11 @@ static FnCallResult FnCallFormat(EvalContext *ctx, ARG_UNUSED const Policy *poli
                         {
                             Writer *w = StringWriter();
```

Comment:
Seems like there is a memory leak here, but it was already there. It's missing a `WriterClose()`.

Reply:
Created separate ticket for this: https://tracker.mender.io/browse/CFE-3528